src/lib/geometry.f90

```diff
@@ -1341,6 +1497,407 @@ subroutine make_data_grid(surface_elems,spacing,to_export,filename,groupname)
       call enter_exit(sub_name,2)
 
       end subroutine make_data_grid
+
+
+!##########################################################################
+
+  subroutine split_datacloud(lobe,to_export)
+  !DEC$ ATTRIBUTES DLLEXPORT,ALIAS:"SO_SPLIT_DATACLOUD" :: SPLIT_DATACLOUD
+
+  !*split_datacloud:* split a point data cloud based on a mesh
+! element
+    use arrays,only: dp,elem_field,elem_nodes,&
+         elems_at_node,num_elems_2d,num_nodes,node_field,field_xyz
+    use arrays,only: data_xyz,num_data
+    use indices,only: ne_a_A,ne_length,ne_radius,ne_vol
+    use mesh_utilities,only: volume_internal_to_surface,point_internal_to_surface
+
+    use diagnostics, only: enter_exit
+    implicit none
+
+     ! Local Variables
+     real(dp) :: max_bound(3),min_bound(3),scale_mesh,translate(3),datalobe_xyz(3,num_data)
+     integer :: surfacelobe_elems(num_elems_2d)
+
+     integer,allocatable :: triangle(:,:), globalnodelobe(:)
+     real(dp),allocatable :: vertex_xyz(:,:)
+     real(dp) :: boxrange(3),cofm_surfaces(3),offset=-2.0_dp,point_xyz(3)
+     integer :: i,j,k,ne,nj,nline,nn,num_data_estimate,num_triangles,num_vertices,numlobe_elems
+     integer :: id,nd,num_datalobe
+     logical :: internal
+
+     logical,intent(in) :: to_export
+     character(LEN=3), intent(in) :: lobe
+
+     character(len=1) :: char1
+     character(len=100) :: writefile,exfile
+     character(len=60) :: sub_name = 'split_datacloud'
+
+    integer,parameter :: num_coords = 3
+
+    !     Local Variables
+    integer :: ierror,np,np1,np2,np_global,surround
+
+    sub_name = 'split_datacloud'
+    call enter_exit(sub_name,1)
+    
+    if(.not.allocated(globalnodelobe)) allocate(globalnodelobe(num_data))
+
+
+     if(lobe == "LLL") then 
+!      surfacelobe_elems = (/65,66,67,71,73,77,78,82,83,101,102,103,&
+!      104,106,107,108,109,111,112,113,114,115,117/) 
```

Comment:
I get the need for hardcoded stuff here - for now :) - but we shouldn't have lines commented out - should be only one option, right?

Reply:
yes . removed all unnecessary comments